src/xpcc/architecture/platform/devices/stm32/stm32l476_486-j_m_q_r_v_z-c_e_g.xml

```diff
@@ -129,7 +129,7 @@
     <driver type="adc" name="stm32l4" instances="1,2,3"/>
     <driver type="can" name="stm32" instances="1"/>
     <driver type="clock" name="stm32"/>
-    <driver type="i2c" name="stm32" instances="1,2,3"/>
+    <driver type="i2c" name="stm32l4" instances="1,2,3"/>
```

Comment:
By visually comparing the I2C register map from all STM32L series

| Reference Manual | Series | Compatible |
| ---------- | ---------- |---|
| RM0038 | STM32L100xx, STM32L151xx, STM32L152xx and STM32L162xx | NO |
| RM0351 | STM32L4x5 and STM32L4x6 | YES |
| RM0377 | STM32L0x1 | YES |
| RM0392 | STM32L4x1 | YES |
| RM0393 | STM32L4x2 | YES |
| RM0394 | STM32L43xxx STM32L44xxx STM32L45xxx STM32L46xxx | YES |
| RM0395 | STM32L4x5 | YES |


I found no obvious difference for STM32L0 and STM32L4. So it seems to be a general rule that the L0 and L4 series uses this 'new' I2C IP. So the name stm32l4 might not be optimal.

The STM32L1 series looks like the STM32F4 but missing the FLTR register.

I updated the DFG without testing. @salkinium Could you please update the device files?